tests/test_web_server.py

```diff
@@ -90,8 +90,9 @@ def handler(request):
     server = yield from raw_test_server(handler, logger=logger)
     cli = yield from test_client(server)
 
-    with pytest.raises(client.ServerDisconnectedError):
-        yield from cli.get('/path/to')
+    resp = yield from cli.get('/path/to')
+    with pytest.raises(client.ClientPayloadError):
```

Comment:
until now, the headers were buffered and not sent to the transport until the first byte of the body was written.

I changed that to send the headers immediately if the transport is ready when `write_headers` is called; it impacts this test because if the server cancels the `write_eof`, the headers are still already sent, so the client does not see the ServerDisconnectedError and instead I have to try to read the body to get a `ClientPayloadError` (the server did not send a complete body)

see the last paragraph of the issue